mantidproject/mantid
# Change sum-banks algorithm to include monitors

This PR modifies the ReflectometryISISSumBanks algorithm to copy the monitors from the original workspace into the result. This is required because the result will be used in the ISIS reflectometry reduction workflow algorithms, which assume the input contains monitors.

Part of #34106

**To test:**

- Run this script:
```
ws = LoadNexusProcessed('INTER45455_inst')
ReflectometryISISSumBanks(InputWorkspace=ws, OutputWorkspace='summed')
```
- Check that the monitors have been copied from the original workspace into the summed workspace, i.e. the first 3 spectra are the same (the spectrum numbers are different, which is expected, but the data is the same):
![image](https://user-images.githubusercontent.com/12895056/177741293-b8c01ced-0789-426a-8845-19302a716b84.png)
- The data should like this:
![image](https://user-images.githubusercontent.com/12895056/177741002-c036c253-dc3a-4e5f-8647-32941cdfb312.png)

*This does not require release notes* because **it concerns a new algorithm that is not in production use yet**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
